Tests/LibWasm/test-wasm.cpp

```diff
@@ -252,15 +252,24 @@ JS_DEFINE_NATIVE_FUNCTION(WebAssemblyModule::wasm_invoke)
     if (result.values().is_empty())
         return JS::js_null();
 
-    JS::Value return_value;
-    result.values().first().value().visit(
-        [&](auto const& value) { return_value = JS::Value(static_cast<double>(value)); },
-        [&](i32 value) { return_value = JS::Value(static_cast<double>(value)); },
-        [&](i64 value) { return_value = JS::Value(JS::BigInt::create(vm, Crypto::SignedBigInteger { value })); },
-        [&](Wasm::Reference const& reference) {
-            reference.ref().visit(
-                [&](const Wasm::Reference::Null&) { return_value = JS::js_null(); },
-                [&](const auto& ref) { return_value = JS::Value(static_cast<double>(ref.address.value())); });
-        });
-    return return_value;
+    auto to_js_value = [&](Wasm::Value const& value) {
+        return value.value().visit(
+            [](auto const& value) { return JS::Value(static_cast<double>(value)); },
+            [](i32 value) { return JS::Value(static_cast<double>(value)); },
+            [&](i64 value) { return JS::Value(JS::BigInt::create(vm, Crypto::SignedBigInteger { value })); },
+            [](Wasm::Reference const& reference) {
+                return reference.ref().visit(
+                    [](const Wasm::Reference::Null&) { return JS::js_null(); },
+                    [](const auto& ref) { return JS::Value(static_cast<double>(ref.address.value())); });
+            });
+    };
+
+    if (result.values().size() == 1)
+        return to_js_value(result.values().first());
+
+    Vector<JS::Value> values;
+    values.ensure_capacity(result.values().size());
+    for (auto& value : result.values())
+        values.append(to_js_value(value));
+    return JS::Array::create_from(*vm.current_realm(), move(values));
```

Comment:
Would be nicer as:

```cpp
auto& realm = *vm.current_realm();
...
return Array::create_from<Wasm::Value>(realm, result.values(), [](auto& value) {
    return to_js_value(value);
});
```

Reply:
Oooh, didn't know about that overload. Sure.